libs/langchain/langchain/document_loaders/parsers/pdf.py

```diff
@@ -244,3 +244,45 @@ def lazy_parse(self, blob: Blob) -> Iterator[Document]:
             page_content=current_text,
             metadata={"source": blob.source, "page": current_page},
         )
+
+
+class DocumentIntelligenceParser(BaseBlobParser):
+    """Loads a PDF with Azure Document Intelligence
+    (formerly Forms Recognizer) and chunks at character level."""
+
+    def __init__(self, model: str, endpoint: str, key: str):
+        self.model = model
+        self.endpoint = endpoint
+        self.key = key
+
+    def _generate_docs(self, blob, result):
+        for p in result.pages:
+            content = " ".join([line.content for line in p.lines])
+
+            d = Document(
+                page_content=content,
+                metadata={
+                    "source": blob.source,
+                    "page": p.page_number,
+                },
+            )
+            yield d
+
+    def lazy_parse(self, blob: Blob) -> Iterator[Document]:
+        """Lazily parse the blob."""
+        from azure.ai.formrecognizer import DocumentAnalysisClient
+        from azure.core.credentials import AzureKeyCredential
+
+        with blob.as_bytes_io() as file_obj:
+            document_analysis_client = DocumentAnalysisClient(
```

Comment:
this can probably be created outside the context manager (doesnt rely on file_obj

in fact - should this even be passed in wholesale to this object? cc @eyurtsev @baskaryan

Reply:
Good point to pass the DocumentAnalysisClient as an instance to the loader. This would remove some complexity (e.g. endpoint/key) from loader and parser. Looks also more similar to how the Textract loader is coded. Will modify and update the PR.